lib/feed/utils/community.dart

```diff
@@ -77,14 +77,36 @@ Future<void> toggleFavoriteCommunity(BuildContext context, Community community,
 }
 
 /// Takes a list of [communities] and returns the list with any [favoriteCommunities] at the beginning of the list
-/// Note that you may need to call [toList] on any lists that are marked as readonly.
+/// Note that you may need to call [toList] when passing in lists that are marked as readonly.
 List<CommunityView>? prioritizeFavorites(List<CommunityView>? communities, List<CommunityView>? favoriteCommunities) {
-  return communities
-    ?..sort(
-      (a, b) => favoriteCommunities?.any((c) => c.community.id == a.community.id) == true
-          ? -1
-          : favoriteCommunities?.any((c) => c.community.id == b.community.id) == true
-              ? 1
-              : b.counts.subscribers.compareTo(a.counts.subscribers),
-    );
+  // This is the list we will return. If we don't do any prioritization,
+  // it will just be the initial list we received.
+  List<CommunityView>? prioritizedCommunities = communities;
```

Comment:
Hmm, I _believe_ there's a less verbose way to do this!

Instead of having to loop through the list in reverse, we can just perform a filter on the original `communities` list to get favorited and non-favorited communities. Once we have those, we can combine them into one. I feel like this makes it a bit easier to understand what's happening.

I wrote up some code to explain what I mean (its untested so not sure if it'll work properly) Let me know what you think!

```dart
// Create a set of the favorited community ids for filtering later
Set<int> favoriteCommunityIds = Set<int>.from(favoriteCommunities!.map((c) => c.community.id));

// Filters out communities that are part of the favorites, and keeps the same order
List<CommunityView> sortedFavorites = communities.where((community) => favoriteCommunityIds.contains(community.id)).toList();

// Filters out communities that are not a part of the favorites, and keeps the same order
List<CommunityView> sortedNonFavorites = communities.where((community) => !favoriteCommunityIds.contains(community.id)).toList();

// Combine them together, with favorites at the top
List<CommunityView> prioritizedCommunities = List<CommunityView>.from(sortedFavorites)..addAll(sortedNonFavorites);
```

Reply:
Sounds good! I had to make a few teaks, but I agree it's _much_ more readable! 😊